src/stravalib/client.py

```diff
@@ -451,26 +451,30 @@ def get_athlete_stats(
 
         return model.AthleteStats.model_validate(raw)
 
-    def get_athlete_clubs(self) -> list[model.Club]:
+    def get_athlete_clubs(
+        self, limit: int | None = None
+    ) -> BatchedResultsIterator[model.SummaryClub]:
         """List the clubs for the currently authenticated athlete.
 
         https://developers.strava.com/docs/reference/#api-Clubs-getLoggedInAthleteClubs
 
+        limit : int or None, default=None
+            Maximum number of club objects to return.
 
         Returns
         -------
-        :class:`list`
-            A list of :class:`stravalib.model.Club`
+        class:`BatchedResultsIterator`
+            An iterator of :class:`stravalib.model.SummaryClub` objects
 
         """
 
-        # TODO: This should return a BatchedResultsIterator or otherwise at
-        # most 30 clubs are returned!
-        club_structs = self.protocol.get("/athlete/clubs")
-        return [
-            model.Club.model_validate({**raw, **{"bound_client": self}})
-            for raw in club_structs
-        ]
+        result_fetcher = functools.partial(self.protocol.get, "/athlete/clubs")
+        return BatchedResultsIterator(
```

Comment:
now returning an iterator following the comment to support >30 clubs (do people have that many clubs?!)